pkg/controllers/dlb/controller.go

```diff
@@ -93,7 +94,7 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 	}
 
 	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
-		ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
+		ds.Spec.Template.Spec.NodeSelector = controllers.AddArchConstraintToNodeSelector(dp.Spec.NodeSelector)
```

Comment:
So far I was under impression that we decided that ```ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector``` is enough for now. Have I missed something?

Reply:
It's a nice idea to have the update logic in a single function that is used by all the controllers. However, I agree with @bart0sh that the current logic is not what was decided: with empty `dp.Spec.NodeSelector` we set the original default value otherwise the use it as is.

Reply:
Bartosh, Eduard Ylinen, Mikko
Sorry that I made something that is not what was decided in the meeting.! As I wrote in github, I lost all of my notes because of this stupid laptop.. I would rather use outlook notes from now on..

Anyway, I thought my memory was wrong because I lost my notes and it seemd to not be possible to set the original default value when the user value is empty.

> ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector

I understand this code is for "replacing" to what user input.

But then,, what is the way to get the "default" value from the yaml?

Or,, do we just put
> map[string]string{"kubernetes.io/arch": "amd64"}
?

Reply:
> But then,, what is the way to get the "default" value from the yaml?

As we added it to all daemonset yamls it will be taken from there by this code: ```daemonSet := deployments.DLBPluginDaemonSet()``` I believe.

Reply:
> Anyway, I thought my memory was wrong because I lost my notes and it seemd to not be possible to set the original default value when the user value is empty.

That's why we had this check:
```
if len(devicePlugin.Spec.NodeSelector) > 0 {
        daemonSet.Spec.Template.Spec.NodeSelector = devicePlugin.Spec.NodeSelector
}
```

Reply:
@bart0sh Thanks!